test/react/helper/use-event-listener-ref.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { renderHook } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+
+import { useEventListenerRef } from '../../../src/react/helper/use-event-listener-ref.js';
+
+describe('useEventListenerRef', () => {
+  it('should update the callback as it changes', () => {
+    const originalCallback = vi.fn();
+    const { result, rerender } = renderHook(({ callback }) => useEventListenerRef(callback), {
+      initialProps: { callback: originalCallback },
+    });
+
+    const initialResult = result.current;
+
+    (result.current as (arg0: string, arg1: string) => void)('arg1', 'arg2');
+    expect(originalCallback).toHaveBeenCalledWith('arg1', 'arg2');
+
+    const newCallback = vi.fn();
+    rerender({ callback: newCallback });
+
+    (result.current as (arg0: string, arg1: string) => void)('arg3', 'arg4');
+    expect(newCallback).toHaveBeenCalledWith('arg3', 'arg4');
+    expect(originalCallback).lastCalledWith('arg1', 'arg2');
+
+    expect(result.current).toBe(initialResult);
+  });
+
+  it('should return handle undefined callbacks ', () => {
```

Comment:
Just a little nitpick, should this be, should handle undefined callbacks?

Reply:
Fixed!